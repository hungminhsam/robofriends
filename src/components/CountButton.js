import React, { useState } from "react";

const CountButton = () => {
  const [count, setCount] = useState(0);

  const onClick = () => {
    setCount((count) => count + 1);
  };

  return <button onClick={onClick}>Click count : {count} </button>;
};

export default CountButton;
