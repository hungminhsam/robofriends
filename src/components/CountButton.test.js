import { shallow } from "enzyme";
import React from "react";
import CountButton from "./CountButton";

describe("<CountButton />", () => {
  it("expect to render CountButton Component", () => {
    expect(shallow(<CountButton />).debug()).toMatchSnapshot();
  });

  it("correctly increments the counter", () => {
    const wrapper = shallow(<CountButton />);
    wrapper.find("button").simulate("click");
    wrapper.find("button").simulate("click");
    expect(wrapper.text()).toEqual("Click count : 2 ");
  });
});
