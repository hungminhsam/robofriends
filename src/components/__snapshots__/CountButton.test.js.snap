// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountButton /> expect to render CountButton Component 1`] = `
"<button onClick={[Function: onClick]}>
  Click count : 
  0
   
</button>"
`;
